### ISSUE TITLE
parted failed with rc 0

### ISSUE THREAD

**Issue by jordipalet** [Drewsif/PiShrink#116]
It looks like everything is going fine, but at the end I get:

Line 287
partnewsize: 11562852352
newpartend: 13550952448
Error: Can't have overlapping partitions.
pishrink.sh: ERROR occured in line 296: parted failed with rc 0

Any hint?

**Comment by situssoft** [comment]
Me too, any idea how to fix this?

**Comment by situssoft** [comment]
Incase anyone else ends up here, for my case of shrinking a raspbian lite based image the script is attempting to create the wrong type of partition on line 307. For my setup the partition type is logical, by changing this line from primary to logical the script completed without error.

**Comment by framps** [comment]
> Incase anyone else ends up here, for my case of shrinking a raspbian lite based image the script is attempting to create the wrong type of partition on line 307. 

Thank you very much for your update.

> For my setup the partition type is logical, by changing this line from primary to logical the script completed without error.

What's special on your setup? If we have to handle this in the code we have to know when to go left and when to go right :wink:

**Comment by situssoft** [comment]
Ok, time to be humble. After further investigation I have realised that this problem is because I had an image createed from NOOBS. The partitioning is entirly different and I did read that this isn't supported yet.

So for clarity. You will get this error if you attempt to shrink an image that is the result of a NOOBS install.

To avoid that, install from the raspbian image (I'm using raspbain lite) and all will be good.
